src/commons/Map.tsx

```diff
@@ -0,0 +1,96 @@
+import React, { useEffect, useRef } from "react";
+
+type Coordinates = {
+  lat: number;
+  lng: number;
+};
+
+type MapProps = {
+  originCoordinates?: Coordinates;
+  destinationCoordinates?: Coordinates;
+};
+
+const Map: React.FC<MapProps> = ({
+  originCoordinates,
+  destinationCoordinates,
+}) => {
+  const mapContainerStyle: React.CSSProperties = {
+    width: "100%",
+    height: "400px",
+  };
+
+  const mapRef = useRef<HTMLDivElement | null>(null);
+  const mapInstanceRef = useRef<google.maps.Map | null>(null);
+  const directionsRendererRef = useRef<google.maps.DirectionsRenderer | null>(
+    null
+  );
+
+  useEffect(() => {
+    const onLoad = () => {
+      if (originCoordinates && destinationCoordinates) {
+        // eslint-disable-next-line @typescript-eslint/no-non-null-assertion
+        const map = new google.maps.Map(mapRef.current!, {
+          center: originCoordinates,
+          zoom: 12,
+        });
+        mapInstanceRef.current = map;
```

Comment:
Quizás este sería un mejor approach
```suggestion
      if (originCoordinates && destinationCoordinates && mapRef.current) {
        const map = new google.maps.Map(mapRef.current, {
          center: originCoordinates,
          zoom: 12,
        });
        mapInstanceRef.current = map;
```

Reply:
okas